managers/bosh-manager/DirectorService.js

```diff
@@ -0,0 +1,1112 @@
+                guid: this.guid
+              }),
+              this.deleteRestoreFile()
+            ]);
+        }
+      });
+  }
+
+  acquireNetworkSegmentIndex(guid) {
+    logger.info(`Acquiring network segment index for a new deployment with instance id '${guid}'...`);
+    const promises = [this.getDeploymentNames(true)];
+    if (config.enable_bosh_rate_limit) {
+      promises.push(this.getDeploymentNamesInCache());
+    }
+    return Promise.all(promises)
+      .then(deploymentNameCollection => _.flatten(deploymentNameCollection))
+      .then(deploymentNames => {
+        const deploymentName = _.find(deploymentNames, name => _.endsWith(name, guid));
+        if (deploymentName) {
+          logger.warn('+-> Deployment with this instance id already exists');
+          throw new ServiceInstanceAlreadyExists(guid);
+        }
+        return NetworkSegmentIndex.findFreeIndex(deploymentNames, this.subnet);
+      })
+      .tap(networkSegmentIndex => logger.info(`+-> Acquired network segment index '${networkSegmentIndex}'`));
+  }
+
+  deleteRestoreFile() {
+    if (_.includes(this.agent.features, 'backup')) {
+      return Promise.try(() => this.platformManager.ensureTenantId({
+          context: this.platformContext,
+          guid: this.guid
+        }))
+        .then(tenant_id => tenant_id ? this.deleteRestoreFileFromObjectStore(tenant_id, this.guid) : Promise.resolve({}))
+        .catch(err => {
+          logger.error(`Failed to delete restore file of instance '${this.guid}'`, err);
+          throw err;
+        });
+    }
+  }
+
+  deleteRestoreFileFromObjectStore(tenant_id, instance_guid) {
+    const options = {
+      tenant_id: tenant_id,
+      service_id: this.service.id,
+      plan_id: this.plan.id,
+      instance_guid: instance_guid
+    };
+    return Promise.try(() => {
+      if (!_.includes(this.agent.features, 'backup')) {
+        return null;
+      } else {
+        return this.backupStore.deleteRestoreFile(options);
+      }
+
+    });
+  }
+
+  finalize(operation) {
+    const action = _.capitalize(operation.type);
+    return Promise
+      .try(() => {
+        switch (operation.type) {
+        case 'create':
+          return utils
+            .retry(tries => {
+              logger.info(`+-> ${ordinals[tries]} attempt to create property '${CONST.PLATFORM_CONTEXT_KEY}' for deployment '${this.deploymentName}'...`);
+              return this.director
+                .createDeploymentProperty(this.deploymentName, CONST.PLATFORM_CONTEXT_KEY, JSON.stringify(operation.context))
+                .catch(err => {
+                  logger.error(`Error occured while trying to create deployment property for deployment ${this.deploymentName}`, err);
+                  throw err;
+                });
+            }, {
+              maxAttempts: 3,
+              minDelay: 1000
+            })
+            .then(() => this.platformManager.postInstanceProvisionOperations({
```

Comment:
usage of platform manager.

Reply:
Platform managers would still remain. SF2.0 managers are for the provisioning use case. Platform managers are for the consuming platform specific use case like SG creation. Cant be removed.